pkg/clusterd/disk.go

```diff
@@ -145,6 +145,9 @@ func PopulateDeviceInfo(d string, executor exec.Executor) (*sys.LocalDisk, error
 	}
 	if val, ok := diskProps["PKNAME"]; ok {
 		disk.Parent = path.Base(val)
+		if disk.Parent == "." {
```

Comment:
What about replacing lines 147-150 with this?
```
if val != "" {
  disk.Parent = path.Base(val)
}
```
But I'm ok with your implementation too, they should have the same effect.

Reply:
good idea. fixed